packages/kit/src/runtime/server/index.js

```diff
@@ -294,7 +283,19 @@ export async function respond(request, options, state) {
 	try {
 		const response = await options.hooks.handle({
 			event,
-			resolve,
+			resolve: (event, opts) =>
+				resolve(event, opts).then((response) => {
```

Comment:
Just from looking at this code I don't quite understand what the difference between this and the previous version is - could we maybe add a small explanatory comment? Is it that `state.prerendering?.fallback` wouldn't get the correct cookies/headers?

Reply:
Headers and cookies were only applied if there was a valid route. So if you used `setHeaders` or `cookies.set` in handle they had no effect

Reply:
It's so that cookies and headers get applied regardless of whether the request happened to match a route (i.e. you call `cookies.set` in `handle` or `src/routes/+layout.server.js` and it still gets applied to a 404 page created with `respond_with_error`). It _will_ apply to fallback responses but that will have no effect, since the fallback response is only ever written out as HTML.

Added a comment

Reply:
jinx